ha/setup/post_disruptive_upgrade.py

```diff
@@ -0,0 +1,173 @@
+    output, _, _ = SimpleCommand().run_cmd("pcs cluster cib")
+    return ElementTree.fromstring(output)
+
+def _check_for_any_resource_presence() -> None:
+    '''Check if any resources are already present in a cluster.
+       if yes, means, pre-upgrade steps failed. hence exit'''
+
+    Log.info('Check for any resource presence in a cluster')
+
+    root = _get_cib_xml()
+    resource_list = [e.attrib["id"] for e in root.findall(".//lrm_resource")
+                if "id" in e.attrib]
+
+    if resource_list:
+        raise UpgradeError('Some resources are already present in the cluster. \
+                            Perform Upgrade process again')
+
+
+def _is_cluster_standby_on() -> None:
+    '''Check if cluster is in standby mode. If not, make standby mode ON'''
+
+    Log.info('Check cluster is in standby mode')
+    value = SimpleCommand().run_cmd(CHECK_PCS_STANDBY_MODE)
+
+    standby_value = value[0].split(' ')[3].strip('\n').split('=')
+
+    if standby_value[1].lower() != 'on':
+        Log.warn('cluster is not in standby mode.')
+        Log.info('switching the cluster in standby mode for performing post upgrade routines')
+        _switch_cluster_mode(PCS_CLUSTER_STANDBY)
+    Log.info('#### All post-upgrade prerequisites are in place ####')
+
+def _restore_consul_backup():
+
+    Log.info('Restoring the consul backup')
+
+def _yaml_to_dict(yaml_file=None):
+    '''
+       Convert yaml format key value info i the form of
+       python dictionary key value
+    '''
+    if yaml_file is None:
+        raise UpgradeError('yaml file path can not be None. Please provide the \
+                 HA yaml conf file path for conversion')
+    with open(yaml_file, 'r') as conf_file:
+        file_as_dict = yaml.safe_load(conf_file)
+    return file_as_dict
+
+def parse_dict(sample_dict, lkey=''):
+    flattened_dict = {}
+    for dict_key, val in sample_dict.items():
+        key = lkey+dict_key
+        if isinstance(val, dict):
+            flattened_dict.update(parse_dict(val, key+':'))
+        else:
+            flattened_dict[key] = val
+    return flattened_dict
+
+
+def _load_config(ha_source_conf: str = SOURCE_CONFIG_FILE, \
+                 ha_backup_conf: str = BACKUP_DEST_DIR_CONF) -> None:
+    '''
+       Load the new config at proper location after the
+       RPM upgrade as part of post-upgrade process
+    '''
+
+    dest_dir = CONFIG_DIR
+    new_src_dir = SOURCE_CONFIG_PATH
+
+    # Convert yaml to dictionary
+    old_backup_conf_dict = _yaml_to_dict(ha_backup_conf)
+    new_conf_dict = _yaml_to_dict(ha_source_conf)
+
+    # Note: There are 3 scenarios for conf file upgrade
+    # 1. New conf key-value pair can be introduced
+    # 2. Already present conf key can be updated with new value
+    # 3. A conf key-value can be deleted
+    # Here, we are considering or assuming thatthere will not be updation.
+    # Upgrade means a new key value will be added. 2nd scenario can be handled
```

Comment:
Is the 2nd scenario handled here? I think the comment should dictate that clearly.

Reply:
No.. Please check now.